### PR TITLE
Added PyCm

### PR DESCRIPTION
**Pull request by rhyspoulton**
This PR adds a recipe for PyCm, which is a Cm library python binding built via ctypegen

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fcarbogn** [comment]
Hi, I am happy to be listed as maintainer for those recipes

**Comment by rhyspoulton** [comment]
@conda-forge/help-python-c, this recipe is ready for review, thanks.

**Comment by fcarbogn** [comment]
We are referring at the moment to the v7r4p5 version. Would be worth putting in conda the last version v7r5p1?
Would be better to put as condition cm >=9.0.0 instead of cm ==9.0.0? 
We discussed this matter in the past at the vdas and we concluded that is always better to put >= everywhere we put constraints in versions

**Comment by rhyspoulton** [comment]
Yes, I agree, I have moved to the latest version of PyCm and updated the constraint on Cm.

**Comment by rhyspoulton** [comment]
Ping @conda-forge/help-python-c, is someone able to review and merge this recipe?

**Comment by rhyspoulton** [comment]
@conda-forge-admin, please ping team

File: recipes/PyCm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{% set name = "pycm" %}
+{% set version = "7.5.1" %}
+{% set tag = "v7r5p1" %}
+
+
+
```
Comment (duncanmmacleod) [inline recipes/PyCm/meta.yaml:6]: Please remove these extra empty lines.

Comment (rhyspoulton) [reply recipes/PyCm/meta.yaml]: Done

File: recipes/PyCm/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	-DCMAKE_INSTALL_PREFIX:PATH=${PREFIX} \
+	-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+;
+
+# build
+cmake --build . --parallel ${CPU_COUNT}
+
+# install
+cmake --build . --parallel ${CPU_COUNT} --target install
+
```
Comment (ocefpaf) [inline recipes/PyCm/build.sh:20]: Is there a make check or ctest?

Comment (rhyspoulton) [reply recipes/PyCm/build.sh:20]: @ocefpaf I have just pushed a commit that adds a `ctest` to the build.

**Comment by rhyspoulton** [comment]
@conda-forge-admin, please ping team

File: recipes/PyCm/build.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+mkdir -p _build
+pushd _build
+
+# configure
+cmake ${SRC_DIR} \
+	-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo \
```
Comment (xhochy) [inline recipes/PyCm/build.sh:8]:
```suggestion
        ${CMAKE_ARGS}
	-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo \
```

File: recipes/PyCm/build.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	-DCMAKE_INSTALL_DATADIR:PATH=${SRC_DIR}/trash \
+	-DCMAKE_INSTALL_LIBDIR:PATH="lib" \
+	-DCMAKE_INSTALL_PREFIX:PATH=${PREFIX} \
```
Comment (xhochy) [inline recipes/PyCm/build.sh:11]: This is part of `${CMAKE_ARGS}`
```suggestion
```

File: recipes/PyCm/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{% set name = "pycm" %}
+{% set version = "7.5.1" %}
+{% set tag = "v7r5p1" %}
```
Comment (duncanmmacleod) [inline recipes/PyCm/meta.yaml:3]: Can you copy what I put together for Frv from [here](https://github.com/conda-forge/frv-feedstock/blob/0fde137a5637c8a1f2ee70c3d3f23c7a68ce2c21/recipe/meta.yaml#L2-L9)?

```suggestion
{% set version = "v7r5p1" %}

# handle Virgo version number format
{% set semver_version = version|
    replace('v', '', 1)|
    replace('r', '.', 1)|
    replace('p', '.', 1)
 %}
```

File: recipes/PyCm/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+   name: "{{ name }}"
+   version: "{{ version }}"
+
+source:
+   url: http://software.igwn.org/lscsoft/source/{{ name }}-{{ version }}.tar.gz
+   sha256: 427466add2f0c3095a4aaece857cd403de603eda1a3462367deae03a4fb27057
+
+build:
+  number: 0
+  skip: true  # [win or osx]
```
Comment (duncanmmacleod) [inline recipes/PyCm/meta.yaml:15]: Does this make more sense?

```suggestion
  skip: true  # [not linux]
```

Comment (rhyspoulton) [reply recipes/PyCm/meta.yaml]: Yes, I agree that makes more sense to put not.

File: recipes/PyCm/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
+{% set name = "pycm" %}
+{% set version = "v7r5p1" %}
+
+# handle Virgo version number format
+{% set semver_version = version|
+    replace('v', '', 1)|
+    replace('r', '.', 1)|
+    replace('p', '.', 1)
+ %}
```
Comment (duncanmmacleod) [inline recipes/PyCm/meta.yaml:9]: Actually, sorry @rhyspoulton, but it strikes me that you're not using the upstream Virgo `tag` at all, so you can just leave this as 

```suggestion
{% set version = "7.5.1" %}
```

Comment (rhyspoulton) [reply recipes/PyCm/meta.yaml]: No worries, yes it makes the most sense just to remove it.

**Comment by duncanmmacleod** [comment]
@conda-forge/staged-recipes, I'm sure someone else has already reported this, but it looks like the osx miniforge installation has just jumped to python3.9, which breaks `networkx` so even `skip: true  # [osx]` builds are breaking [[ref]](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=308965&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310):

```pytb
+ python .ci_support/build_all.py

Traceback (most recent call last):
  File "/Users/runner/work/1/s/.ci_support/build_all.py", line 2, in <module>
    import networkx as nx
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/networkx/__init__.py", line 114, in <module>
    import networkx.generators
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/networkx/generators/__init__.py", line 14, in <module>
    from networkx.generators.intersection import *
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/networkx/generators/intersection.py", line 13, in <module>
    from networkx.algorithms import bipartite
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/networkx/algorithms/__init__.py", line 16, in <module>
    from networkx.algorithms.dag import *
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/networkx/algorithms/dag.py", line 23, in <module>
    from fractions import gcd
ImportError: cannot import name 'gcd' from 'fractions' (/Users/runner/miniforge3/lib/python3.9/fractions.py)
```

This was corrected in https://github.com/networkx/networkx/pull/3639 and released with `networkx-2.4`.

**Comment by jakirkham** [comment]
@conda-forge-admin, please restart CI

**Pull request merged**